substrate/frame/support/src/lib.rs

```diff
@@ -869,7 +869,7 @@ macro_rules! hypothetically_ok {
 pub use serde::{Deserialize, Serialize};
 
 #[doc(hidden)]
-#[cfg(not(no_std))]
+#[cfg(feature = "std")]
```

Comment:
```suggestion
```

I think this was a no-op previously and causes errors if you guard it now. CI should tell.

Reply:
but `no_std` is defined in frame-support with this line:
```
#![cfg_attr(not(feature = "std"), no_std)]
```

Reply:
Yea its unfortunate, as we *should not* need it in WASM, but no clue why it gets pulled in 🤷‍♂️

Reply:
maybe when we call in this:
```
		#[frame_support::register_default_impl(SolochainDefaultConfig)]
```
the registration is done in both std and no std. And it uses macro magic in the expansion somewhere